.github/workflows/ci.yml

```diff
@@ -0,0 +1,185 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+    lint-shellcheck:
+        runs-on: ubuntu-latest
+        steps:
+          - uses: actions/checkout@v2
+
+          - name: Runs shell script static analysis
+            run: |
+              sudo apt-get install shellcheck
+              ./run-tests.sh --check-shellscript
+
+    lint-black:
+        runs-on: ubuntu-latest
+        steps:
+          - uses: actions/checkout@v2
+
+          - name: Setup Python
+            uses: actions/setup-python@v2
+            with:
+              python-version: 3.8
+
+          - name: Install Python dependencies
+            run: pip install black==19.10b0
+
+          - name: Check Python code formatting
+            run: ./run-tests.sh --check-black
+
+    lint-pydocstyle:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Setup Python
+          uses: actions/setup-python@v2
+          with:
+            python-version: 3.8
+
+        - name: Install Python dependencies
+          run: pip install pydocstyle
+
+        - name: Check compliance with Python docstring conventions
+          run: ./run-tests.sh --check-pydocstyle
+
+    lint-check-manifest:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Setup Python
+          uses: actions/setup-python@v2
+          with:
+            python-version: 3.8
+
+        - name: Install Python dependencies
+          run: pip install check-manifest
+
+        - name: Check Python manifest completeness
+          run: ./run-tests.sh --check-manifest
```

Comment:
I see we are approaching this a bit different than in e.g. [RWC GH Actions PR](https://github.com/reanahub/reana-workflow-controller/pull/347/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR57-R72). Maybe this has been discussed before and I didn't see the outcome of the conversation?

Reply:
It's the same, I just separated the `installation` in a different step.